client/config.c

```diff
@@ -26,6 +29,50 @@ TDNFConfGetRpmVerbosity(
     return nLogLevel;
 }
 
+static uint32_t TDNFParseOSInfo(PTDNF_CONF pConf)
+{
+    char temp[USERAGENT_HEADER_MAX_LENGTH];
```

Comment:
I would still say we should avoid static allocation here and use getline. It's upto you on what you want to use.

Also, be aware that fgets doesn't always null terminate the buffer. By chance if there is a line which is more than 50 bytes in os-release file, this might end up crashing tdnf.

Reply:
That's not what the map page says:
```
fgets()  reads in at most one less than size characters from stream and stores them into the buffer pointed to by s.  Reading stops after an EOF or a newline.  If a newline is read, it is stored into
       the buffer.  A terminating null byte ('\0') is stored after the last character in the buffer.
```
Instead it seems to take care that it is aways terminated with 0.